src/rdkafka_cgrp.c

```diff
@@ -5154,8 +5189,42 @@ rd_kafka_cgrp_subscription_set(rd_kafka_cgrp_t *rkcg,
                     rkcg->rkcg_rk, rkcg->rkcg_subscription, NULL,
                     0 /*dont replace*/);
         }
+
+        if (rkcg->rkcg_group_protocol == RD_KAFKA_GROUP_PROTOCOL_CONSUMER) {
+                RD_IF_FREE(rkcg->rkcg_subscription_topics,
+                           rd_kafka_topic_partition_list_destroy);
+                RD_IF_FREE(rkcg->rkcg_subscription_regex,
+                           rd_kafkap_str_destroy);
+
+                if (rkcg->rkcg_subscription) {
+                        rkcg->rkcg_subscription_topics =
+                            rd_kafka_topic_partition_list_copy(
+                                rkcg->rkcg_subscription);
+
+                        if (is_wildcard_subscription_present) {
+                                rkcg->rkcg_subscription_regex =
+                                    rd_kafka_topic_partition_list_combine_regexes(
+                                        rkcg->rkcg_subscription);
+                                rd_kafka_topic_partition_list_remove_regexes(
```

Comment:
You can make `rd_kafka_topic_partition_list_remove_regexes` directly return the filtered topic partition list and assign it to `rkcg->rkcg_subscription_topics` here